paasta_tools/vitesscluster_tools.py

```diff
@@ -1,27 +1,544 @@
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-secretid",
+                "key": "vault-vttablet-approle-secretid",
+            }
+        },
+    },
+]
+
+# Vault auth related variables
+VTGATE_EXTRA_ENV = [
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-roleid",
+                "key": "vault-vtgate-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-secretid",
+                "key": "vault-vtgate-approle-secretid",
+            }
+        },
+    },
+]
+
+
+# Extra Flags
+VTADMIN_EXTRA_FLAGS = {"grpc-allow-reflection": "true"}
+
+VTCTLD_EXTRA_FLAGS = {
+    "disable_active_reparents": "true",
+    "security_policy": "read-only",
+}
+
+VTTABLET_EXTRA_FLAGS = {
+    "log_err_stacks": "true",
+    "grpc_max_message_size": "134217728",
+    "init_tablet_type": "replica",
+    "queryserver-config-schema-reload-time": "1800",
+    "dba_pool_size": "4",
+    "vreplication_heartbeat_update_interval": "60",
+    "vreplication_tablet_type": "REPLICA",
+    "keep_logs": "72h",
+    "enable-lag-throttler": "true",
+    "throttle_check_as_check_self": "true",
+    "db_charset": "utf8mb4",
+    "disable_active_reparents": "true",
+}
+
+
+def get_updated_environment_variables(original_list: List[Dict], update_dict: Dict):
+    for env in original_list:
+        if env["name"] in update_dict:
+            env["value"] = update_dict[env["name"]]
+            del update_dict[env["name"]]
+    original_list.extend([{"name": k, "value": v} for k, v in update_dict.items()])
+    return original_list
+
+
+def get_affinity_spec(paasta_pool: str) -> Dict[str, Dict[str, List[str]]]:
+    spec = {
+        "nodeAffinity": {
+            "requiredDuringSchedulingIgnoredDuringExecution": {
+                "nodeSelectorTerms": [
+                    {
+                        "matchExpressions": [
+                            {
+                                "key": "yelp.com/pool",
+                                "operator": "In",
+                                "values": [paasta_pool],
+                            }
+                        ]
+                    }
+                ]
+            }
+        }
+    }
+    return spec
+
+
+def get_extra_labels(paasta_pool: str, paasta_cluster: str) -> Dict[str, str]:
+    """
+    get extra labels to adhere to paasta contract
+    """
+    extra_labels = {
+        "yelp.com/owner": "dre_mysql_working_hours",
+        "paasta.yelp.com/cluster": paasta_cluster,
+        "paasta.yelp.com/pool": paasta_pool,
+    }
+    return extra_labels
+
+
+def get_extra_env(paasta_cluster: str) -> List[Dict[str, str]]:
```

Comment:
(as-is: making a change to the base classes wouldn't update things for vitess - which probably isn't the end of the world, but it will make things more manual since it'd be easy to forget to update the vitess code)

Reply:
and i guess another question is: as-is, i don't think there'd be a way for folks to add in an `env` key for these instances to add custom env vars: is that fine (or is it just late and i'm mistaken :p)?

Reply:
(actually, i think if many of these functions became helpers in `VitessDeploymentConfig` we could maybe also cut down the number of params to some of these functions)

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/48ebe609e02377336248dcbb9bfed29a0d6213ae